wicket-core/src/main/java/org/apache/wicket/markup/html/image/ExternalImage.java

```diff
@@ -0,0 +1,295 @@
+import org.apache.wicket.markup.html.WebComponent;
+import org.apache.wicket.markup.html.image.Image.Cors;
+import org.apache.wicket.model.IModel;
+import org.apache.wicket.model.Model;
+
+/**
+ * A component to display external images. The src / srcset information are hold in models
+ * 
+ * @author Tobias Soloschenko
+ *
+ */
+public class ExternalImage extends WebComponent
+{
+
+	private static final long serialVersionUID = 1L;
+
+	/** The x values to be used within the srcset */
+	private List<String> xValues = null;
+
+	/** The sizes of the responsive images */
+	private List<String> sizes = null;
+
+	/**
+	 * Cross origin settings
+	 */
+	private Cors crossOrigin = null;
+
+	private IModel<?>[] srcSetModels;
+
+	/**
+	 * Creates an external image
+	 * 
+	 * @param id
+	 *            the component id
+	 * @param src
+	 *            the source URL
+	 * @param srcSet
+	 *            a list of URLs placed in the srcset attribute
+	 */
+	public ExternalImage(String id, Serializable src, Serializable... srcSet)
+	{
+		this(id, src != null ? Model.of(src) : null, convertToModel(srcSet));
+	}
+
+	/**
+	 * Creates an external image
+	 * 
+	 * @param id
+	 *            the component id
+	 * @param srcModel
+	 *            the model source URL
+	 * @param srcSetModels
+	 *            a model list of URLs placed in the srcset attribute
+	 */
+	public ExternalImage(String id, IModel<?> srcModel, IModel<?>... srcSetModels)
+	{
+		super(id, srcModel);
+		this.srcSetModels = srcSetModels;
+	}
+
+	/**
+	 * Converts a variable argument of Strings to an array of models
+	 * 
+	 * @param srcSet
+	 *            a variable argument of URLs to be converted to an array of models
+	 * @return an array of models
+	 */
+	private static IModel<?>[] convertToModel(Serializable... srcSet)
+	{
+		IModel<?>[] models = null;
+		if (srcSet != null)
+		{
+
+			models = new IModel<?>[srcSet.length];
+			int i = 0;
+			for (Serializable srcSetElement : srcSet)
+			{
+				models[i] = Model.of(srcSetElement);
+				i++;
+			}
+		}
+		else
+		{
+			models = new IModel<?>[0];
+		}
+		return models;
+	}
+
+	@Override
+	protected void onComponentTag(ComponentTag tag)
+	{
+		super.onComponentTag(tag);
+
+		if ("source".equals(tag.getName()))
+		{
+			buildSrcSetAttribute(tag, getSrcSet());
+		}
+		else
+		{
+			List<IModel<?>> srcSet = getSrcSet();
+			checkComponentTag(tag, "img");
+			buildSrcAttribute(tag, getDefaultModel());
+			if (srcSet.size() > 1)
+			{
+				buildSrcSetAttribute(tag, srcSet);
+			}
+		}
+
+		buildSizesAttribute(tag);
+
+		Cors crossOrigin = getCrossOrigin();
+		if (crossOrigin != null && Cors.NO_CORS != crossOrigin)
+		{
+			tag.put("crossOrigin", crossOrigin.getRealName());
+		}
+	}
+
+	/**
+	 * Builds the src attribute
+	 *
+	 * @param tag
+	 *            the component tag
+	 * @param srcModel
+	 *            the model containing the src URL
+	 */
+	protected void buildSrcAttribute(final ComponentTag tag, IModel<?> srcModel)
+	{
+		if (srcModel != null)
```

Comment:
IMO, you better have to use `#getDefaultModelObject()` because the model supplied in arg can be null, whereas the model can exists through the component hierarchy...